Egui image api

melody-rs opened the pull request:
**Description**
egui `0.23.0` got a new image API, and deprecated `egui_extras::RetainedImage`. This pull request adds a custom egui image loader to load from the project filesystem.

This allows us to use `ui.image("project://{})"` to display images from the currently open project in a way that aligns with egui's design.

<!-- 
Thanks for filing a pull request! The codeowners file will automatically request reviews.

After you get a review and have addressed any comments, please explicitly re-request a review from the
person(s) who reviewed your changes. This will make sure it gets re-added to their review queue - you're not bothering us!
-->

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
- [x] Run `cargo build --release` 
- [x] If applicable, run `trunk build --release`

melody-rs committed:
chore: use universal set of lints across crates

melody-rs committed:
chore: :rotating_light: auto-fix most warnings

melody-rs committed:
chore: disable panic and unwrap lints for now

melody-rs committed:
chore: :rotating_light: fix wasily fixable warnings

melody-rs committed:
refactor(image cache): :recycle: add basic egui texture loader

\

melody-rs committed:
fix(image cache): :bug: handle image load errors semi-gracefully

melody-rs committed:
fix(image cache): :bug: clear up the difference between uris and paths

melody-rs committed:
refactor(image cache): :recycle: drop the use of multiple bind groups for one monolithic bind group

melody-rs committed:
refactor(graphics): :recycle: remove pervasive passing of push constant support

melody-rs committed:
fix(tilemap): :bug: fix various shader and bind group issues

melody-rs committed:
refactor(tilemap): :recycle: don't pass viewports around constantly

melody-rs committed:
fix(tilemap): :bug: fix the collision shader crashing

melody-rs committed:
refactor: :recycle: remove all uses of retainedimage

melody-rs committed:
fix(image cache): :bug: fix loading errors & textures not being srgb

melody-rs committed:
fix(image cache): :bug: fix srgb weirdness

melody-rs committed:
fix(image cache): :bug: properly unload textures

white-axe committed:
Add CJK font (#71)

* feat: add Source Han Sans (for displaying CJK characters)

* perf: compress fonts to reduce binary size

melody-rs committed:
refactor(image cache): :fire: remove TextureLoader and add BytesLoader instead

melody-rs committed:
refactor: :recycle: Rename atlas cache to atlas loader

melody-rs committed:
fix: :rotating_light: fix compile error on wasm

melody-rs committed:
Merge branch 'dev' into egui-image-api

melody-rs committed:
rustfmt

melody-rs committed:
Merge branch 'dev' into egui-image-api

white-axe reviewed (changes requested):
I get this wgpu error when enabling the collision shader on native (web is fine):
```
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `encoder`
    In a draw command, indexed:false indirect:false
      note: render pipeline = `Tilemap Collision Render Pipeline`
    The pipeline layout, associated with the current render pipeline, contains a bind group layout at index 0 which is incompatible with the bind group layout associated with the bind group at 0
```

melody-rs committed:
fix(tilemap): :bug: fix collision shader on native

I don't know how I missed this, but the collision shader still expected there to be a bind group when we weren't passing one!

melody-rs commented:
Should be fixed now!
